lib/kube/proxy/forwarder.go

```diff
@@ -1459,6 +1611,28 @@ func (f *Forwarder) setupForwardingHeaders(sess *clusterSession, req *http.Reque
 
 // setupImpersonationHeaders sets up Impersonate-User and Impersonate-Group headers
 func setupImpersonationHeaders(log logrus.FieldLogger, ctx authContext, headers http.Header) error {
+	impersonateUser, impersonateGroups, err := computeImpersonatedPrincipals(log, ctx.kubeUsers, ctx.kubeGroups, headers)
```

Comment:
Start with the remote check and exit early?

```suggestion
	if ctx.teleportCluster.isRemote {
		return nil
	}
	impersonateUser, impersonateGroups, err := computeImpersonatedPrincipals(log, ctx.kubeUsers, ctx.kubeGroups, headers)
```

then drop the if and indent below.